pkg/provisioner/ironic/ironic.go

```diff
@@ -974,8 +991,33 @@ func (p *ironicProvisioner) Provision(hostConf provisioner.HostConfigData) (resu
 		return p.startProvisioning(ironicNode, checksum, hostConf)
 
 	case nodes.Manageable:
+		// manual clean
+		result, err = p.startManualCleaning(ironicNode)
+		// wait clean finish
```

Comment:
Everywhere else when the provisioner makes a state change in ironic we let it return and the Reconcile() function in the controller exits, to be called again after a pause. One reason we need to do that is that all of the reconcile logic needs to be idempotent in case the pod containing the baremetal-operator is restarted. A restart can happen automatically during a system upgrade or if the pod is evicted from its host.

Is there a way to make this code idempotent and avoid a sleep loop here?

Reply:
Thanks for your suggestions, I have modified the code.